src/Components/Server/src/Circuits/RemoteRenderer.RootComponentOperations.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits;
+
+// Contains logic for root components that get dynamically updated in Blazor Web scenarios.
+internal partial class RemoteRenderer
+{
+    private Dictionary<int, ParameterView>? _latestRootComponentDirectParameters;
```

Comment:
There's also `_rootComponentsLatestParameters` in `Renderer.cs`, but that's only used/updated for hot reload scenarios. I want to limit the tracking of root component parameters to precisely the scenarios that need it.